### PR TITLE
connman: 1.37 -> 1.38

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

- Update connman 1.38, see [change log](https://git.kernel.org/pub/scm/network/connman/connman.git/commit/?id=1ee420ace2b8edb0d4025f469aaa3d00d220dc98).
- Cleanup build for easy enable / disable of features.

###### Things done

- [X] Rewrite all features' enable / disable flags.
- [X] Cleaned up dependencies, including unneeded ones.
- [X] Added 2 new build flavors - `connmanFull` and `connmanThin` with several more / less features enabled / disabled.
- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @matejc

### CODE REVIEW

**Comment by worldofpeace** [comment]
Perhaps @jtojnar or @romildo (I think he was interested in connman) could look at this.

**Review by romildo — APPROVED** [review]
This PR looks good to me. The many options are well organized.

I have tested on my notebook with wifi and ethernet connections.

Maybe an option could be added to the `connman` service to select the `connman` package, as there will be three to choose from: `connman`, `connmanMinimal` and `connmanFull`.

**Comment by doronbehar** [comment]
That's definitely a good idea @romildo . I created an option for the package and I think it works - tested with my own `nixos-config`. I've added some TODOs regarding network-manager+connman, see last commit. BTW This my first contribution to nixos' modules :).

File: nixos/modules/services/networking/connman.nix [hunk 1]
```diff
@@ -89,11 +96,13 @@ in {
       assertion = !config.networking.useDHCP;
       message = "You can not use services.connman with networking.useDHCP";
     }{
+      # TODO: connman seemingly can be used along network manager and
+      # connmanFull supports this - so this should be worked out somehow
       assertion = !config.networking.networkmanager.enable;
       message = "You can not use services.connman with networking.networkmanager";
     }];
```
Comment (worldofpeace) [inline nixos/modules/services/networking/connman.nix]: can you also open an issue? We have a tracker for these things.

**Review by worldofpeace — APPROVED** [review]
All looks good for to me. An issue would be nice for that todo.

File: nixos/modules/services/networking/connman.nix (hunk 1 above)
Comment (doronbehar) [reply nixos/modules/services/networking/connman.nix]: @worldofpeace it's just that I personally never have used or tried to use network manager at all. It'll take me quiet an effort to write a good bug report so I think it'd be proper to wait for someone else to want that feature. Hopefully he'll encounter this comment indicating at least someone was aware of this..

**Pull request merged**